### PR TITLE
feat/DCMAW-8814: Create HelloWorld sts call

### PR DESCRIPTION
**Pull request by alex-bradbury**
- Creation of `AuthTabScreen` to live in the developer menu. This has two buttons to allow a happy and sad path call to `/hello-world` endpoint
- `AuthTabScreenViewModel` handles making the calls to hello world. Sad call temporarily deletes the stored access token to simulate an error
- STS endpoint feature is turned on
- Use case `HelloWorldApiCall` created to handle calls to `/hello-world` endpoint
- Necessary hilt modules created to provide http clients from the android-network module
- Slight changes in `WelcomeScreenViewModel` to fix calling the STS `/authorize` url

Resolves: DCMAW-8814

### CODE REVIEW

File: app/src/main/java/uk/gov/onelogin/developer/tabs/auth/AuthTabScreenViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    fun makeHappyHelloWorldCall() {
+        _happyCallLoading.value = true
+        viewModelScope.launch {
+            _happyHelloWorldResponse.value = helloWorldApiCall()
+            _happyCallLoading.value = false
+        }
+    }
+
+    fun makeSadHelloWorldCall() {
```
Comment (flavius-mester) [inline app/src/main/java/uk/gov/onelogin/developer/tabs/auth/AuthTabScreenViewModel.kt:42]: This works!

What I was having in mind was to perhaps modify the other Tab. `SecureStor` and add a button there `Purge token from Runtime`, `Restore token to Runtime`, `Purge token from Disk` etc.

Just to the QA(and us/other devs) has visibility of why the network request fails.

**Review by flavius-mester — COMMENTED** [review]
Looking good, left some comments

File: app/src/main/java/uk/gov/onelogin/developer/tabs/auth/AuthTabScreenViewModel.kt (hunk 1 above)
Comment (alex-bradbury) [reply app/src/main/java/uk/gov/onelogin/developer/tabs/auth/AuthTabScreenViewModel.kt:42]: mmmm true, think that functionality would be nice but maybe too much to add to this 'ticket'

File: app/src/main/java/uk/gov/onelogin/network/di/NetworkModule.kt [hunk 2]
```diff
@@ -30,15 +33,34 @@ object NetworkModule {
...
+        val appName = context.resources.getString(R.string.app_name)
+        userAgentGenerator.setUserAgent(
+            UserAgent(
+                appName = appName,
+                versionName = BuildConfig.VERSION_NAME,
+                clientName = KTOR_CLIENT,
+                manufacturer = Build.MANUFACTURER,
+                model = Build.MODEL,
+                sdkVersion = Build.VERSION.SDK_INT,
+                clientVersion = BuildConfig.KTOR_VERSION
```
[the remark this answers is not in the document]
Comment (alex-bradbury) [reply app/src/main/java/uk/gov/onelogin/network/di/NetworkModule.kt:48]: yeah good point, looked at what Wallet were doing but actually very much agree, will change to app version

**Pull request merged**